src/main/java/edu/ie3/datamodel/io/csv/FileHierarchy.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * © 2020. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.io.csv;
+
+import edu.ie3.datamodel.models.UniqueEntity;
+import java.util.Optional;
+
+/**
+ * Abstract definition of a file hierarchy (a piece of software, that knows where to find / place a
+ * file reflecting a certain class) inside of a nested sub directory structure
+ */
+public interface FileHierarchy {
+  /**
+   * Determines the correct subdirectory (w.r.t. an arbitrary base directory) for a certain given
+   * class using the provided file separator for delimiting between directories and files.
+   *
+   * @param cls Class to define the sub directory for
+   * @param fileSeparator The file separator to use
+   * @return An Option to the regarding sub directory as a string
+   */
+  Optional<String> getSubDirectory(Class<? extends UniqueEntity> cls, String fileSeparator);
+
+  /**
+   * Determines the correct subdirectory (w.r.t. an arbitrary base directory) for a certain given
+   * class using the Unix file separator for delimiting between directories and files.
+   *
+   * @param cls Class to define the sub directory for
+   * @return An Option to the regarding sub directory as a string
+   */
+  default Optional<String> getSubDirectory(Class<? extends UniqueEntity> cls) {
+    return getSubDirectory(cls, "/");
```

Comment:
Again hard coded file separator - I would expected issues here? Why not using `File.separator();`